File: backend/src/main/java/com/ssafy/libro/domain/book/entity/Book.java
package com.ssafy.libro.domain.book.entity;

import com.ssafy.libro.domain.book.dto.BookUpdateRequestDto;
import com.ssafy.libro.domain.userbook.entity.UserBook;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String isbn;
    private String title;

    @Column(columnDefinition = "LONGTEXT")
    private String summary;
    private Integer price;
    private Double rating;
    private Integer ratingCount;

    private String author;
    private String translator;
    private String publisher;
    private LocalDateTime pubDate;

    private String thumbnail;
    private String shortsUrl;


    @CreationTimestamp
    private LocalDateTime createdDate;
    @UpdateTimestamp
    private LocalDateTime updatedDate;

    //join
    @OneToMany(mappedBy = "book", fetch = FetchType.LAZY, targetEntity = UserBook.class)
    private List<UserBook> userBookList;

    public Book update(BookUpdateRequestDto requestDto) {
        this.isbn = requestDto.getIsbn();
        this.title = requestDto.getTitle();
        this.rating = requestDto.getRating();
        this.summary = requestDto.getSummary();
        this.author = requestDto.getAuthor();
        this.translator = requestDto.getTranslator();
        this.publisher = requestDto.getPublisher();
        this.thumbnail = requestDto.getThumbnail();
        this.shortsUrl = requestDto.getShortsUrl();
        return this;
    }

    public Book updateRating(Double rating) {
        this.rating = rating;
        return this;
    }

}

File: recommend/data/book.py
class Book : 
    id: int
    title : str
    author : str
    translator : str
    isbn : str
    price : int
    pub_date : str
    publisher : str
    rating : float
    rating : int
    shorts_url: str
    summary : str
    thumbnail : str
    created_date : str
    update_date : str
    
    def __init__(self, **kwargs):
        for key, value in kwargs.items():
            setattr(self, key, value)
    


    def to_dict(self):
        """
        Book 클래스의 객체를 딕셔너리로 변환하는 함수
        """
        book_dict = {}
        for key, value in self.__dict__.items():
            if not key.startswith("__") and not callable(getattr(self, key)):
                book_dict[key] = value
        return book_dict
    # def __init__(self, id: int, title: str, author: str, translator: str, isbn: str,
    #              price: int, pub_date: str, publisher: str, rating: float,
    #              rating_count: int, shorts_url: str, summary: str, thumbnail: str,
    #              created_date: str, update_date: str):
    #     self.id = id
    #     self.title = title
    #     self.author = author
    #     self.translator = translator
    #     self.isbn = isbn
    #     self.price = price
    #     self.pub_date = pub_date
    #     self.publisher = publisher
    #     self.rating = rating
    #     self.rating_count = rating_count
    #     self.shorts_url = shorts_url
    #     self.summary = summary
    #     self.thumbnail = thumbnail
    #     self.created_date = created_date
    #     self.update_date = update_date
